packages/atomic/src/components/search/atomic-result-timespan/atomic-result-timespan.spec.ts

```diff
@@ -0,0 +1,247 @@
+      (result: Result, property: string) => {
+        return result.raw[property] ?? null;
+      }
+    );
+  });
+
+  const renderResultTimespan = async ({
+    props = {},
+    result,
+  }: {
+    props?: Partial<{field: string; unit: string; format: string}>;
+    result?: Result;
+  } = {}) => {
+    const resultToUse = result ?? mockResult;
+    const {element, atomicInterface} =
+      await renderInAtomicResult<AtomicResultTimespan>({
+        template: html`<atomic-result-timespan
+          field=${ifDefined(props.field)}
+          unit=${ifDefined(props.unit)}
+          format=${ifDefined(props.format)}
+        ></atomic-result-timespan>`,
+        selector: 'atomic-result-timespan',
+        result: resultToUse,
+        bindings: (bindings) => {
+          bindings.i18n = i18n;
+          return bindings;
+        },
+      });
+
+    await atomicInterface.updateComplete;
+    await element?.updateComplete;
+
+    return {element, atomicInterface};
+  };
+
+  describe('when validating props', () => {
+    it('should not set error when field is a non-empty string', async () => {
+      const {element} = await renderResultTimespan({
+        props: {field: 'duration'},
+      });
+
+      expect(element.error).toBeUndefined();
+    });
+
+    it('should set error when field is an empty string', async () => {
+      const {element} = await renderResultTimespan({props: {field: ''}});
+
+      expect(element.error).toBeInstanceOf(Error);
+    });
+
+    it('should set error when field is undefined', async () => {
+      const {element} = await renderResultTimespan();
+
+      expect(element.error).toBeInstanceOf(Error);
+    });
+  });
+
+  describe('when initializing', () => {
+    it('should not set error when field value exists and is a number', async () => {
+      vi.mocked(ResultTemplatesHelpers.getResultProperty).mockReturnValue(
+        3600000
+      );
+
+      const {element} = await renderResultTimespan({
+        props: {field: 'duration'},
+      });
+
+      expect(element.error).toBeUndefined();
+    });
+
+    it('should set error when field value does not exist', async () => {
+      vi.mocked(ResultTemplatesHelpers.getResultProperty).mockReturnValue(null);
+
+      const {element} = await renderResultTimespan({props: {field: 'missing'}});
+
+      expect(element.error).toBeInstanceOf(Error);
+      expect(element.error.message).toBe('No value found for field missing');
+    });
+
+    it('should set error when field value is NaN', async () => {
+      vi.mocked(ResultTemplatesHelpers.getResultProperty).mockReturnValue(NaN);
+
+      const {element} = await renderResultTimespan({
+        props: {field: 'invalid'},
+        result: buildFakeResult({raw: {invalid: NaN}}),
+      });
+
+      expect(element.error).toBeInstanceOf(Error);
+      expect(element.error.message).toBe('No value found for field invalid');
+    });
+  });
+
+  describe('when rendering with HH:mm:ss format (durations under a day)', () => {
```

Comment:
I think in this case it would be worth testing some "edge case" scenarios (e.g., 0 values, negative values...)

The thing is the data ultimately comes from a field in the index, so it's basically "user input", and it could be completely wrong. We need to ensure that the component handles all of these cases gracefully.